### PR TITLE
[formatting-only] De-tabify .java files

### PR DESCRIPTION
**Pull request by josh-hadley**
### Reviewer Notes

Tabs in .java files were replaced with either 4 or 8 spaces, with one exception: in [unicodetools/src/main/java/org/unicode/text/UCD/CompareProperties.java](https://github.com/unicode-org/unicodetools/pull/178/files#diff-70ba9d508fcb94220008c63d27753cc5760fc74edfa9859ca9a03abeb7bd0a7dL443) there were un-escaped tabs inside of a string of an `output.println()` statement. These were replaced with escaped tabs (`"\t"`) rather than spaces, to match the output of other `output.println()` statements using escaped tabs.

### CODE REVIEW

File: unicodetools/src/main/java/org/unicode/text/UCD/CompareProperties.java [hunk 1]
```diff
@@ -440,7 +440,7 @@ public static void listDifferences() throws IOException {
         output.println("# Listing of relationships among properties, suitable for analysis by spreadsheet");
         output.println("# Generated for " + Default.ucd().getVersion());
         output.println(Utility.generateDateLine());
-        output.println("# P1	P2	R(P1,P2)	C(P1&P2)	C(P1-P2)	C(P2-P1)");
+        output.println("# P1\tP2\tR(P1,P2)\tC(P1&P2)\tC(P1-P2)\tC(P2-P1)");
```
Comment (srl295) [inline unicodetools/src/main/java/org/unicode/text/UCD/CompareProperties.java:443]: good

**Review by srl295 — DISMISSED** [review]
lgtm

File: unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiData.java [hunk 2]
```diff
@@ -45,740 +45,740 @@
...
-	    } else if (Emoji.GENDER_MARKERS.containsSome(s)) {
-		zwjType = gestures;
-	    }
-	    return zwjType;
-	}
+    roleWithHair, roleWithObject, roleWithSign, gestures, activity, family, other, na;
+    public static ZwjType getType(String s) {
+        if (!s.contains(Emoji.JOINER_STRING)) {
+        return na;
+        }
```
Comment (markusicu) [inline unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiData.java:118]: This file also wanted 8sp/tab.

File: unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiTestFile.java [hunk 3]
```diff
@@ -271,7 +271,7 @@ public static void showLines(EmojiOrder emojiOrder, UnicodeSet charactersToInclu
                 for (String cp : vf.getCombinations()) {
                     charactersNotShown.remove(cp);
                     VariantStatus variantStatus = emojiOrder.emojiData.getVariantStatus(cp);
-		    String ageDisplay = "E" + BirthInfo.getVersionInfo(cp).getVersionString(2, 2);
+            String ageDisplay = "E" + BirthInfo.getVersionInfo(cp).getVersionString(2, 2);
```
Comment (markusicu) [inline unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiTestFile.java:274]:
```suggestion
                    String ageDisplay = "E" + BirthInfo.getVersionInfo(cp).getVersionString(2, 2);
```

File: unicodetools/src/test/java/org/unicode/propstest/CheckGCB.java [hunk 4]
```diff
@@ -33,283 +33,283 @@
...
-	    Word_Break_Values.Regional_Indicator
-	    );
+        Word_Break_Values.Regional_Indicator
+        );
 
     public static void main(String[] args) {
         final Segmenter gcb =
                 Segmenter.make(IUP, "GraphemeClusterBreak", Segmenter.Target.FOR_UCD).make();
-	final UnicodeMap<UcdPropertyValues.Grapheme_Cluster_Break_Values> gcbProps = IUP.loadEnum(UcdProperty.Grapheme_Cluster_Break);
+    final UnicodeMap<UcdPropertyValues.Grapheme_Cluster_Break_Values> gcbProps = IUP.loadEnum(UcdProperty.Grapheme_Cluster_Break);
```
Comment (markusicu) [inline unicodetools/src/test/java/org/unicode/propstest/CheckGCB.java:44]: This file wanted 8sp/tab.

File: unicodetools/src/test/java/org/unicode/propstest/TestXSet.java [hunk 5]
```diff
@@ -18,108 +18,108 @@
...
-			}
-			break;
-		    }
-		    // ∪ ∩
-		}
-	    }
-	}
+    String[] tests = {
+        "[]", "[a]", "[b]", "[c]", "[ab]", "[bc]", "[ca]", "[abc]",
+    };
```
Comment (markusicu) [inline unicodetools/src/test/java/org/unicode/propstest/TestXSet.java:26]: This file wanted 8sp/tab.

File: unicodetools/src/test/java/org/unicode/tools/emoji/unittest/TestEmojiData.java [hunk 6]
```diff
@@ -595,37 +595,37 @@ public void TestBuckets(String type, EmojiDataSource emojiDataToTest, EmojiOrder
...
-		}
-	    }
-	    System.out.println("Discarded: " + discarded);
+        final EmojiData released = EmojiData.of(Emoji.VERSION_BETA);
+        Multimap<String,String> toRgi = TreeMultimap.create();
+        UnicodeSet discarded = new UnicodeSet();
+        for (String item : released.getAllEmojiWithoutDefectives()) {
+        String alphagram = getAlphagram(item, released, discarded);
+        toRgi.put(alphagram, item);
+        }
```
Comment (markusicu) [inline unicodetools/src/test/java/org/unicode/tools/emoji/unittest/TestEmojiData.java:604]: This file wanted 8sp/tab.

**Review by markusicu — CHANGES_REQUESTED** [review]
thanks!

File: unicodetools/src/main/java/org/unicode/tools/FixNcrs.java [hunk 7]
```diff
@@ -11,36 +11,36 @@
 public class FixNcrs {
     // TODO add command line support later
     public static void main(String[] args) throws IOException {
-//	String relativeFileName = FileUtilities.getRelativeFileName(FixNcrs.class, "entity.txt/../");
-//	TransliteratorUtilities.registerTransliteratorFromFile(relativeFileName, "entity");
-	Transliterator te = Transliterator.getInstance("hex-any/xml; hex-any/xml10");
+//    String relativeFileName = FileUtilities.getRelativeFileName(FixNcrs.class, "entity.txt/../");
+//    TransliteratorUtilities.registerTransliteratorFromFile(relativeFileName, "entity");
+    Transliterator te = Transliterator.getInstance("hex-any/xml; hex-any/xml10");
```
[the remark this answers is not in the document]
Comment (josh-hadley) [reply unicodetools/src/main/java/org/unicode/tools/FixNcrs.java]: Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiData.java (hunk 2 above)
Comment (josh-hadley) [reply unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiData.java]: Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiTestFile.java (hunk 3 above)
Comment (josh-hadley) [reply unicodetools/src/main/java/org/unicode/tools/emoji/GenerateEmojiTestFile.java]: Updated in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/test/java/org/unicode/propstest/CheckGCB.java (hunk 4 above)
Comment (josh-hadley) [reply unicodetools/src/test/java/org/unicode/propstest/CheckGCB.java]: Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/test/java/org/unicode/propstest/TestXSet.java (hunk 5 above)
Comment (josh-hadley) [reply unicodetools/src/test/java/org/unicode/propstest/TestXSet.java]: Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/test/java/org/unicode/tools/emoji/unittest/TestEmojiData.java (hunk 6 above)
Comment (josh-hadley) [reply unicodetools/src/test/java/org/unicode/tools/emoji/unittest/TestEmojiData.java]: Fixed in https://github.com/unicode-org/unicodetools/pull/178/commits/7c5c10bdc0232617b772ea6dd0e1e9389a3a9cfc

File: unicodetools/src/main/java/org/unicode/tools/emoji/Emoji.java [hunk 8]
```diff
@@ -101,51 +101,51 @@
...
-	YEAR_TO_EMOJI_VERSION_ASCENDING = ImmutableMap.copyOf(_map);
-	EMOJI_VERSION_TO_YEAR = ImmutableMap.copyOf(_mapEmojiToYear);
+    Map<Integer,VersionInfo> _map = new TreeMap<>();
+    Map<VersionInfo,Integer> _mapEmojiToYear = new TreeMap<>();
+    for (Entry<VersionInfo, String> entry : EMOJI_TO_DATE.entrySet()) {
+        int year = Integer.parseInt(entry.getValue().substring(0, 4));
+        _mapEmojiToYear.put(entry.getKey(), year);
+        if (!_map.containsKey(year)) {
+        _map.put(year, entry.getKey());
+        }
```
Comment (markusicu) [inline unicodetools/src/main/java/org/unicode/tools/emoji/Emoji.java:145]: This file does not look right. Appears to want 8sp/tab.

**Review by markusicu — COMMENTED** [review]
just one more file

File: unicodetools/src/main/java/org/unicode/tools/emoji/Emoji.java (hunk 8 above)
Comment (josh-hadley) [reply unicodetools/src/main/java/org/unicode/tools/emoji/Emoji.java]: Updated in https://github.com/unicode-org/unicodetools/pull/178/commits/aac5aa8e8c617d550104f1d3cd6ba6fd687950ff

**Review by markusicu — APPROVED** [review]
lgtm, thanks!!

please **squash & merge**

**Pull request merged**